core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/StreamBatchSource.java

```diff
@@ -94,7 +96,9 @@
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     super.configurePipeline(pipelineConfigurer);
     streamBatchConfig.validate();
-    pipelineConfigurer.addStream(new Stream(streamBatchConfig.name));
+    if (!streamBatchConfig.containsMacro(Properties.Stream.NAME)) {
+      pipelineConfigurer.addStream(new Stream(streamBatchConfig.name));
```

Comment:
@albertshau @chtyim should we allow this in configure or not ? since we are not allowing stream creation with stream names from macro at runtime. would it make sense to disallow this too ?